astropy/tests/test_logger.py

```diff
@@ -215,7 +215,7 @@ def test_exception_logging():
         sys.excepthook(*sys.exc_info())
         assert exc.args[0] == "This is an Exception"
     else:
-        assert False  # exception should have been raised
+        raise AssertionError()  # exception should have been raised
```

Comment:
Why is the code here not using `with pytest.raises()`?

Reply:
I think only one of the tests can be easily simplified because pytest manipulates the log, erasing the exception from the log history.

Reply:
Done the one that can be done.